index.js

```diff
@@ -33,176 +33,383 @@ const app = express()
+    // Checks if current state belongs to the session triggerStates
+    else if (TRIGGERSTATES.includes(currentState)) {
+      const client = await db.beginTransaction()
+      const latestSession = await db.getMostRecentSession(currentLocationId, client)
+
+      if (typeof latestSession !== 'undefined') {
+        // Checks if session exists
+        if (latestSession.end_time == null) {
+          // Checks if session is open for this location
+          const currentSession = await db.updateSessionState(latestSession.sessionid, currentState, client)
+          start_times[currentLocationId] = currentSession.start_time
+        } else {
+          const currentSession = await db.createSession(location.phonenumber, currentLocationId, currentState, client)
+          start_times[currentLocationId] = currentSession.start_time
+        }
+      } else {
+        const currentSession = await db.createSession(location.phonenumber, currentLocationId, currentState, client)
+        start_times[currentLocationId] = currentSession.start_time
+      }
+      await db.commitTransaction(client)
+    }
+
+    // Checks if current state belongs to the session closingStates
+    else if (CLOSINGSTATES.includes(currentState)) {
+      const client = await db.beginTransaction()
+
+      const latestSession = await db.getMostRecentSession(currentLocationId, client)
+      await db.updateSessionState(latestSession.sessionid, currentState, client)
+
+      await db.closeSession(latestSession.sessionid, client)
+      helpers.log(`Session at ${currentLocationId} was closed successfully.`)
+      start_times[currentLocationId] = null
+      await db.commitTransaction(client)
+    } else if (ALERTSTARTSTATES.includes(currentState)) {
+      const latestSession = await db.getMostRecentSession(currentLocationId)
+
+      // eslint-disable-next-line eqeqeq
+      if (latestSession.od_flag == 1) {
+        if (latestSession.chatbot_state === null) {
+          const alertInfo = {
+            sessionId: latestSession.sessionid,
+            toPhoneNumber: location.phonenumber,
+            fromPhoneNumber: location.twilio_number,
+            message: `This is a ${latestSession.alert_reason} alert. Please check on the bathroom at ${location.display_name}. Please respond with 'ok' once you have checked on it.`,
+            reminderTimeoutMillis: location.unresponded_timer,
+            fallbackTimeoutMillis: location.unresponded_session_timer,
+            reminderMessage: `This is a reminder to check on the bathroom`,
+            fallbackMessage: `An alert to check on the bathroom at ${location.display_name} was not responded to. Please check on it`,
+            fallbackToPhoneNumber: location.fallback_phonenumber,
+            fallbackFromPhoneNumber: location.twilio_number,
+          }
+          braveAlerter.startAlertSession(alertInfo)
+        }
+      }
+    } else {
+      helpers.log('Current State does not belong to any of the States groups')
     }
-}));
```

Comment:
In this ridiculously long file, I'm not sure what these brackets were there for. Is it OK that they've been taken out now?

Reply:
After investigating, it looks like this is now split over a couple of lines to add trailing commas

```
app.use(session({
    key: 'user_sid',
    secret: helpers.getEnvVar('SECRET'),
    resave: false,
    saveUninitialized: false,
    cookie: {
        expires: 10*60*1000
    }
}));
```
has become:
```
app.use(
  session({
    key: 'user_sid',
    secret: helpers.getEnvVar('SECRET'),
    resave: false,
    saveUninitialized: false,
    cookie: {
      expires: 10 * 60 * 1000,
    },
  }),
)
```

Reply:
I'm glad you asked though because I had a momentary panic since this was initially aligned with a line corresponding to `handleSensorRequest` in the review preview and I did move that beast of a function 😆

Reply:
Thanks for finding that!